lib/https.js

```diff
@@ -100,6 +100,8 @@ function createConnection(port, host, options) {
     options = {};
   }
 
+  options = { ...options };
```

Comment:
nit: this will make a copy when we know the object is empty (last conditional above).

Perhaps we could rewrite the conditionals to be something like:

```js
  if (port !== null && typeof port === 'object')
    options = { ...port };
  else if (host !== null && typeof host === 'object')
    options = { ...host };
  else if (options === null || typeof options !== 'object')
    options = {};
  else
    options = { ...options };
```

Reply:
Right. I will work on that. Just out of curiosity: how does using spread operator empty object affect the performance? One thing that I can think of is maybe it won't hamper the performance much by itself but it may add up with several other minute perf degraders and overall make the performance down. Correct me if I am wrong.

Reply:
Right. I will work on that. Just out of curiosity: how does using spread operator empty object affect the performance? One way that I can think of is maybe it won't hamper the performance much by itself but it may add up with several other minute performance degraders and overall make the performance down. Correct me if I am wrong.

Reply:
IMO it's best to play it safe with changes like this (especially when it's an easy enough change) because it's not always safe to make assumptions about V8 optimizations. It could be that V8 is smart enough to avoid performance issues at all, or it could be a significant performance hit by itself, or it could be a smaller performance hit that adds up with any other performance hits (as you mentioned).

Additionally, it's further complicated by the fact that optimizations (or lack thereof) can change across versions of V8.